Revise: improve Windows .ico (multi-image icon files)

SlySven opened the pull request:
The "dev" one needed some reworking to avoid some clipping and I found the GIMP script to generate a new one given a 256x256 starting image I applied the same script to produce new files (with more different images sizes and formats) for the other two as well.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

![image](https://user-images.githubusercontent.com/6163092/94326424-b5015a80-ff9b-11ea-82d4-1ce7e2977d22.png) ==> ![image](https://user-images.githubusercontent.com/6163092/94326384-6f449200-ff9b-11ea-9c88-05d4016d5e1e.png)

SlySven committed:
Revise: improve Windows .ico (multi-image icon files)

The "dev" one needed some reworking to avoid some clipping and I found the
GIMP script to generate a new one given a 256x256 starting image I applied
the same script to produce new files (with more different images sizes and
formats) for the other two as well.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: type in CMake project file

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: I typo-ed the fix for the typo!

Lemme at some coffee...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (commented):
I like the old version better...

SlySven commented:
Well look at how the ellipse of lit ground on the threshold will be sliced off by the bottom of the icon - the other two don't do that - and to be fair to @Kebap (who I think generated the icon because I didn't at the time) it probably escaped his notice. If it would help I ought to be able to make the colour of that patch a bit sharper, more opaque and a tad reddish (it isn't as solid as perhaps it could be in my suggestion).  Would that more likely to meet with approval?

vadi2 commented:
Ah lets see what others think, I don't want to drag this out so see the general opinion and make a call that way

Kebap commented:
> @Kebap (who I think generated the icon because I didn't at the time) it probably escaped his notice. 

Indeed I did but hardly touched any brushes myself, instead standing on the shoulders of giants merely de-/activated different levels in the combined image file, then transformed the result into files that could actually be used as icon.

Only difference I see spontaneously is how you moved the yellow oval more central below the arch. Am I missing something else? Change seems fine to me.

Kebap commented on src/CMakeLists.txt:
```diff
@@ -30,7 +30,13 @@ if(USE_FONTS)
 
 if(WIN32)
-  list(APPEND mudlet_RCCS icons/mudlet_main_512x512_6XS_icon.ico)
+  if(BUILD matches "-ptb.+")
+    list(APPEND mudlet_RCCS icons/mudlet_ptb.ico)
+  elseif((BUILD matches "-ptb.+") or (BUILD matches "-test.+"))
```
For some reason, this checks for "ptb" twice?

SlySven replied:
Ah, yeah the first should have been `"-dev.+"`...

SlySven replied:
Fixed now.

Kebap commented on src/mudlet.pro:
```diff
@@ -1433,14 +1433,18 @@ macx {
 
 win32 {
-    # set the Windows binary icon
+    # set the Windows binary icon, the proper .ico file should contains several
+    # layers of increasing sizes: 16, 24, 32, 48, 64, 128, 256 - as well as
+    # being used for the executable application file when it is displayed in
+    # the Windows explorer it is also used in some other places (installed
+    # applications in Windows settings/control panel)
```
Comment creep explaining basic icon files' dynamics. 
Is that really needed in Mudlet source?
Isn't there a wiki or something?

SlySven replied:
In the *mudlet/installer* repository we have a `./windows/mudlet_main_48px.ico` which is a single 48x48 image (and a `./windows/mudlet_main_512x512px.ico`).

That 48x48 icon is being used for the Windows `setup.exe` currently - and that DOES need revising.

🤔 TBH I don't tend to go to the Wiki whilst hacking the codebase - but it would be useful to store these details somewhere.

vadi2 replied:
Agree on moving this out, perhaps good to start making use of the wiki?

SlySven replied:
Done - and moved to https://wiki.mudlet.org/w/Notes_on_QMake_variables .

vadi2 reviewed (commented):
People like the change so it's good with me.

Only comment is didn't you lower the resolution? The old one was max 512 which looks good on HiDPI screens, new max is 256.

SlySven commented:
> Only comment is didn't you lower the resolution? The old one was max 512 which looks good on HiDPI screens, new max is 256.

Oops, I'd better check my source material as I thought 256 was the highest resolution I had for all the images.  Agreed that 512 would be better if I have it...

SlySven commented:
Working on this... https://github.com/Mudlet/artwork/pull/4 has added high resolution (1024x1024) dev and ptb icons so I can use them as a starting point. As it happens I have also generated the range of `.png`s needed for MacOs `icns` icons as well  for those two variants - however I am lacking a means to actually convert them into that latter file format myself - and I am trying to find a way https://softwarerecs.stackexchange.com/q/76327/69601 ....

SlySven commented:
> Only comment is didn't you lower the resolution? The old one was max 512 which looks good on HiDPI screens, new max is 256.

According to all the information I have found, for the Windows .ICO format only one icon (the first) in the file is to be a 256x256 32-bit compressed png (all the remaining ones are smaller sizes stored without compression) - and, with the right script that is what you get - there is no indication that larger size images are actually supported:
![image](https://user-images.githubusercontent.com/6163092/95940859-4dd70900-0dd7-11eb-985e-55b00964976e.png)
Sadly, the "Help" for this feature has still not been written...

SlySven committed:
Enhance: import high resolution .PNGs from artwork repository

They have been rebuilt from separately extracted elements from the original
SVG file and then stored at 1024x1024 resolution in a GIMP XCF file in
the mudlet/artwork repository. Have pulled them into this repo so they
can be used to make replacement Windows ICO icon files that are stored in
the ./src/icons sub-directory.

They might be needed for other really high definition icons and suchlike as
well.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: change some image save parameters to reduce file size

I had selected some non-optimum (max compression, interlacing) options
that could be fixed without compromising anything.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Enhance: replace/upgrade mudlet Windows .ico icon files

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: fix a couple of icon setting details

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_improveWidowsPTBIcons

vadi2 commented:
Sure, but our 512 icon worked without known issues before, so why downgrade?

vadi2 reviewed (commented):
You are right that Windows right now doesn't make use of more than 256 in an icon file - but it allows for it, so why not future-proof ourselves?

SlySven committed:
Revise: trim a lot of data from a comment.

Suggested by peer-review - excess details transferred to:
https://wiki.mudlet.org/w/Notes_on_QMake_variables

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Enhance_improveWidowsPTBIcons

SlySven committed:
BugFix: insert missing double-quotes

It seems they may be needed when using a back-slash as a directory
separator in a file pathname

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
